src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/action/SnapshotActionConfig.kt

```diff
@@ -57,6 +59,21 @@ data class SnapshotActionConfig(
         managedIndexMetaData: ManagedIndexMetaData
     ): Action = SnapshotAction(clusterService, client, managedIndexMetaData, this)
 
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
+        repository = sin.readOptionalString(),
```

Comment:
Confused why repo and snapshot are nullable in the primary constructor and then we have an init that requires them not to be null, perhaps it was just missed in a previous review. Can you change these to not be nullable so you don't have to do read optional?

Reply:
good point, so I just remove optional then, will leave this `String? and require(repository != null)` combination, looks informative haha

Reply:
will do require in parse